Simplify unit tests runner

ethervoid opened the pull request:
Signed-off-by: Mario de Frutos <mario@defrutos.org>

<!--  Thanks for sending a pull request!  Before submitting:

1. Read our CONTRIBUTING.md guide
2. Rebase your PR if it gets out of sync with master
-->

**What this PR does**:

This PR simplifies how the test tool works reducing significantly the unit test time

**Which issue(s) this PR fixes**:
Fixes #3579 
**Checklist**
- [x] Tests updated
- [ ] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

ethervoid commented:
@pracucci I this the approach you had in mind for this issue?

pracucci reviewed (commented):
Thanks for proposing this! The major slowdown is introduced by `-race -covermode=atomic` which has been removed in this PR. However, removing `-race` also lowers the confidence we get in unit tests so I'm not sure we should remove it.

pracucci commented:
> @pracucci I this the approach you had in mind for this issue?

The idea in the issue was completely removing the `tools/test` and just running `go test <cli flags>` in Makefile, while keeping the same CLI flags (except for the coverage which is something I believe we don't really need in CI).

ethervoid commented:
> The idea in the issue was completely removing the tools/test and just running go test <cli flags> in Makefile, while keeping the same CLI flags (except for the coverage which is something I believe we don't really need in CI).

I see, didn't want to be aggressive removing the file, mainly because of the coverage part. I'll remove that test tool and just use the plain old test CLI.

THx!

ethervoid commented:
@pracucci I've made the changes:

- Removed the `/tools/test` script
- Modified the test task in the Makefile
- Added a new `cover` task to the Makefile just in case is needed by someone in the future

I've checked and now is taking 10 minutes when the old script was taking 24 minutes. Mainly because we've removed the coverage part:

#### New

![Screenshot from 2021-04-30 12-53-47](https://user-images.githubusercontent.com/741240/116685919-5e398480-a9b3-11eb-89ac-2935f41160ce.png)

### Old
![Screenshot from 2021-04-30 12-53-42](https://user-images.githubusercontent.com/741240/116685920-5ed21b00-a9b3-11eb-82f7-f29808444d12.png)

pracucci commented on Makefile:
```diff
@@ -200,7 +200,14 @@ lint:
 
 test:
-	./tools/test -netgo
+	go test -tags netgo -timeout 30m -race -count 1 ./...
```
You missed `-covermode=atomic` which was specified in `tools/test`.

```suggestion
	go test -tags netgo -timeout 30m -race -covermode=atomic -count 1 ./...
```

ethervoid replied:
I don't think we need the covermode here because we're not doing coverage in the test target and if we set `covermode` we're enabling the cover part for the tests as you can see [here](https://github.com/golang/go/blob/06ac303f6a14b133254f757e54599c48e3c2a4ad/src/cmd/go/internal/test/test.go#L229)

On the other hand in the cover target is being used because is the default mode in case we're using race as you can see [here](https://github.com/golang/go/blob/06ac303f6a14b133254f757e54599c48e3c2a4ad/src/cmd/go/internal/test/test.go#L222-L223)

pracucci replied:
Oh you're right. All good here, thanks!

pracucci reviewed (commented):
LGTM (modulo a fix). Thanks!

pracucci reviewed (approved):
LGTM, thanks!

pstibrany reviewed (approved):
Thank you!
